test/fashion-mnist/infer.py

```diff
@@ -0,0 +1,115 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchvision
+import torchvision.transforms as transforms
+
+# Define the neural network model
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.fc1 = nn.Linear(784, 256)
+        self.fc2 = nn.Linear(256, 128)
+        self.fc3 = nn.Linear(128, 10)
+
+    def forward(self, x):
+        x = x.view(x.size(0), -1)
+        x = torch.relu(self.fc1(x))
+        x = torch.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+## IMPROVE
+from improve import framework as frm
+import candle
+from pathlib import Path
+
+filepath = Path(__file__).resolve().parent
+
+
+# Part 3: Model Testing
+## IMPROVE
+def run(params):
+##
+
+   # Need to get testloader from Part 1.
+   transform = transforms.Compose([
+      transforms.ToTensor(),
+      transforms.Normalize((0.5,), (0.5,))
+   ])
+   
+   # Get the data directory, batch size and other hyperparameters from params 
+   ##IMPROVE 
+   batch_size = params["batch_size"]
+   learning_rate = params["learning_rate"]
```

Comment:
Why are these parameters for inference?

Reply:
@rajeeja I agree with @wilke, it's weird that these parameters are defined in inference script. Generally, inference is done without knowledge of training settings (e.g., train batch size, learning rate optimizer, etc.). Is there a reason why these are defined here?

Reply:
@adpartin @wilke 
1. we need to load the test data in batches -> Hence - batch_size (see line 50)
2. I'm getting the model from the ckpt method, we can get it via other means also. The ckpt object needs optimizer to instantiate -> Hence - optimizer (see line 64)

Reply:
Sorry, but it is unclear why you must load the data for inference in batches. Is this in any way faster than a simple
```
for v in File
  label=infer(v)
```

Reply:
There should be a better way of loading the model. We want the model from the input directory. No optimizer is needed. If this is a problem, I suggest writing a load_model_weights function as a wrapper.

Reply:
@rajeeja @adpartin If we use this as an example, we have to make it clean. This is a great hack but not a sustainable solution. Please come up with a better solution or hide it in a function call. These are constants in this case.